### ISSUE TITLE
Enable annotations for the (symfony) validator

### ISSUE THREAD

**Issue by Falseee** [silexphp/Silex#1588]
Hello,

im using silex ^2.2 (symfony validator ^3.3).
I dont understand how i can enable the annotations for the validator..
I found the symfony page https://symfony.com/doc/3.4/validation.html with the example code
`$container->loadFromExtension('framework', array(
    'validation' => array(
        'enable_annotations' => true,
    ),
));`

But i dont understand how i can do it on silex because there is no "loadFromExtension" method.
Btw. im using only php to configure silex (monolog, swiftmailer, etc.)

I hope you can help me, i hope i understand then how i can "execute" the symfony docu on silex.

Greetings,
Falseee

**Comment by hkdobrev** [comment]
We're using https://github.com/saxulum/saxulum-validator-provider which does that. The code doing that is quite simple: https://github.com/saxulum/saxulum-validator-provider/blob/32f6ce87a05ccdfd12f6381365b89aa2612ece48/src/Provider/SaxulumValidatorProvider.php#L27-L33

**Comment by Falseee** [comment]
@hkdobrev 
Why i need a new provider ?
The Validator works with the loadValidatorMetadata function, but not with annotation.
I can not enable the annotations without a new provider ?

**Comment by hkdobrev** [comment]
@Falseee If you don't want to install that provider, you configure the metadata loader yourself, that's why I've pointed you to the code doing that - it's simple enough if you want to configure it yourself.

**Comment by Falseee** [comment]
@hkdobrev 
Ohhhh okay i was to concentrated to the provider, so i forgot about the easiest solution.

**Comment by Falseee** [comment]
@hkdobrev 
I installed the provider, but now i get a error
```
[Semantical Error] The annotation "@Assert\Choice" in property Application\Entity\User\User::$username was never imported. Did you maybe forget to add a "use" statement for this annotation?
```
But i imported it...
```
use Symfony\Component\Validator\Constraints\Choice;
```

**Comment by Falseee** [comment]
no one?

**Comment by hkdobrev** [comment]
I don't see any evident problem with your `use` here. Please double check your code, autoloader configuration etc. Anyway, this is not an issue with Silex and support is probably better suited for StackOverflow.

**Comment by hkdobrev** [comment]
The root reason of why another provider is need would probably be addressed in #1484 which adds the option to enable annotation reading using the built-in `ValidatorServiceProvider`.

**Comment by fabpot** [comment]
It's not built-in (and should probably not be included in core) because of the Silex's philosophy: https://github.com/silexphp/Silex/wiki/Silex-Philosophy

That helps draw the line between Silex, a micro-framework, and Symfony, especially version 4. So, I'm closing as a won't fix (and if a third-party provides support, then that's great).
